### ISSUE TITLE
Can`t delete row in dynamicRows component

### ISSUE THREAD

**Issue by MTheProgrammer** [magento/magento2#8830]
### Preconditions
1. Magento 2.1.4

### Steps to reproduce

1. Go to Products -> Catalog
2. Edit any product
3. Click Add Attribute button
4. Create new attribute of type Visual Swatch or Text Swatch
5. Add swatch
6. Try to Delete row with Bin icon

### Expected result

1. Row is deleted

### Actual result

An error is produced in the js console output
```
Uncaught TypeError: $data.processingDeleteRecord is not a function
    at UiClass.eval (eval at createBindingsStringEvaluator (knockout.js:2624), <anonymous>:3:83)
    at HTMLButtonElement.<anonymous> (knockout.js:3863)
    at HTMLButtonElement.dispatch (jquery.js:4624)
    at HTMLButtonElement.elemData.handle (jquery.js:4292)
```

### Solution
In Catalog/view/adminhtml/web/template/form/element/action-delete.html

update `click: function(){ $data.processingDeleteRecord($parents); },`
to `click: function(){ $parent.processingDeleteRecord($record().index, $record.recordId); },`

**Comment by dmitriyprime** [comment]
Hi, @MTheProgrammer . An internal ticked MAGETWO-70336 was created based on Your issue. You'll be informed as the issue is fixed. Thanks for applying.

**Comment by magento-engcom-team** [comment]
@MTheProgrammer, thank you for your report.
We've created internal ticket(s) MAGETWO-70336 to track progress on the issue.

**Comment by RomaKis** [comment]
Working on this.

**Comment by magento-team** [comment]
Hi @MTheProgrammer. Thank you for your report. 
The issue has been fixed in magento-engcom/magento2ce#921 by @RomaKis in 2.2-develop branch
Related commit(s):
 - 0a6a2d9b80859261afc01c3027f06babdec878aa
 - 53c1fa324eb73394ff5edf02e3cd8c3ee8f23dab
 - 1ab55ec53228ac6183124eb54031e7a6bdf2ec99
 - cb241157d2405d3b7ca0cd3e36c8399d26d2888d
 - bfb2e382b76df806f3a3307b1552dce28b6fa1eb
 - 568d36e3d11ba4633ece4a0969680360bdf4707b
 - 0a7289714f03276cccb1c0fbf860d85c08153abb

The fix will be available with the upcoming patch release.
